addons/account/models/account_move.py

```diff
@@ -4130,11 +4185,13 @@ def write(self, vals):
                     raise UserError(_('You cannot modify the taxes related to a posted journal item, you should reset the journal entry to draft to do so.'))
 
             # Check the lock date.
-            if any(self.env['account.move']._field_will_change(line, vals, field_name) for field_name in PROTECTED_FIELDS_LOCK_DATE):
+            if not line.move_id.is_sale_document(include_receipts=True) \
+                    and any(self.env['account.move']._field_will_change(line, vals, field_name) for field_name in PROTECTED_FIELDS_LOCK_DATE):
```

Comment:
Same, check that the move is not posted.

Also, there is not a lot of recommendations in PEP8, but avoiding `\` to continue a line is one of them (it just makes a mess in the indentation)
https://www.python.org/dev/peps/pep-0008/#maximum-line-length